containers/rust/.devcontainer/devcontainer.json

```diff
@@ -1,34 +1,37 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the README at:
+// https://github.com/microsoft/vscode-dev-containers/tree/v0.158.0/containers/rust
 {
 	"name": "Rust",
 	"build": {
 		"dockerfile": "Dockerfile"
 	},
-	"runArgs": [ "--cap-add=SYS_PTRACE", "--security-opt", "seccomp=unconfined" ],
-
+	"runArgs": [
+		"--cap-add=SYS_PTRACE",
+		"--security-opt",
+		"seccomp=unconfined"
+	],
 	// Set *default* container specific settings.json values on container create.
-	"settings": { 
+	"settings": {
 		"terminal.integrated.shell.linux": "/bin/bash",
 		"lldb.executable": "/usr/bin/lldb",
 		// VS Code don't watch files under ./target
 		"files.watcherExclude": {
 			"**/target/**": true
-		}
+		},
+		"rust-analyzer.checkOnSave.command": "clippy"
```

Comment:
Should we add the notation of `rust-analyzer` for giving a choice to the user? Just like `forwardPorts` below, and I searched for `clippy`. And I found many people have other views https://internals.rust-lang.org/t/should-rust-analyzer-run-clippy-by-default/12766, Would you mind explaining more about this? I'd like to know more about this on my personal.

Reply:
Fine by me to add the notation of `rust-analyzer` for giving a choice to the user. 

As for `clippy` vs`check`, `cargo clippy` itself runs `cargo check` but also gives additional warnings. Using as much warnings as possible generally results in more stable and better code. Though, I would agree that making it the default is opinionated.

Reply:
Thank you